### PR TITLE
refactor cpu func unit tests

### PR DESCRIPTION
**Pull request by xipingyan**
This issue will be splitted into 2 PR,  This is first.

### Details:
 - *Moving cpuFuncTests from src/tests  to src/plugins/intel_cpu/tests/functional*
 - *Moving cpuUnitTests from src/tests to src/plugins/intel_cpu/tests/unit*

### Tickets:
 - *73613*

### CODE REVIEW

**Comment by xipingyan** [comment]
Only move tests code. 
Don't update codes, except for add "inline" for some functions declaration to fix build errors.

File: src/plugins/intel_cpu/CMakeLists.txt [hunk 1]
```diff
@@ -37,6 +37,10 @@ if(TBB_FOUND AND TBB_VERSION VERSION_GREATER_EQUAL 2018 AND TBB_VERSION VERSION_
 endif()
 
 add_subdirectory(thirdparty)
+add_subdirectory(tests/unit_tests)
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/CMakeLists.txt:40]: Shouldn't both subfolders be wrapped by `ENABLE_TESTS` ?

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp [hunk 2]
```diff
@@ -140,7 +140,7 @@ inline bool supportsAvaliableDevices(ov::Core& ie, const std::string& target_dev
            std::find(std::begin(supported_properties), std::end(supported_properties), ov::available_devices);
 }
 
-bool supportsDeviceID(ov::Core& ie, const std::string& target_device) {
+inline bool supportsDeviceID(ov::Core& ie, const std::string& target_device) {
```
Comment (dmitry-gorokhov) [inline src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp:143]: I don't my in fact, but just wondering why we need this change?

**Comment by dmitry-gorokhov** [comment]
@xipingyan Could you please keep folder names: `tests/func_tests` -> `tests/functional` and `tests/unit_tests` -> `tests/unit`. Otherwise we have kind of duplication in paths.

File: src/plugins/intel_cpu/CMakeLists.txt (hunk 1 above)
Comment (xipingyan) [reply src/plugins/intel_cpu/CMakeLists.txt:40]: Yes.

**Review by ilyachur — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]
Please resolve the conflict

**Comment by xipingyan** [comment]
> Please resolve the conflict

Rebased.

**Review by wangleis — APPROVED** [review]

**Pull request merged**